tests/common/ConfigurationXUnit.cs

```diff
@@ -47,11 +47,41 @@ public override IEnumerable<object []> GetData (MethodInfo testMethod)
 		}
 	}
 
+	[AttributeUsage (AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+	public sealed class AllSupportedPlatformsAttribute : DataAttribute {
+
+		readonly object [] dataValues;
+
+		public AllSupportedPlatformsAttribute (params object [] parameters)
+		{
+			dataValues = parameters;
+		}
+
+		public override IEnumerable<object []> GetData (MethodInfo testMethod)
+		{
+			foreach (var platform in Configuration.GetAllPlatforms (true)) {
+				if (!Configuration.IsEnabled (platform))
+					continue;
+				var parameters = dataValues.Prepend (platform).ToArray ();
+				yield return parameters;
+			}
```

Comment:
This is even simpler:

```suggestion
			return Configuration.
				GetIncludedPlatforms (true).
				Select (v => dataValues.Prepend (platform).ToArray ());
```